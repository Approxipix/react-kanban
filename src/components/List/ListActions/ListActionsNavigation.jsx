import React, { Component } from 'react';
import ListActionsNavigationList from "./ListActionsNavigationList";
import styled from 'styled-components'

const List = styled.ul`
`;

const Item = styled.li`
  cursor: pointer;
`;

const Link = styled.a`
  display: block;
  padding: .3rem .75rem;
  font-weight: 700;
  transition: all .2s ease-in;
  &:hover {
    color: #fff;
    background-color: #2E7EAF;
  }
`;
Link.displayName = 'Link';

class ListActionsNavigation extends Component {
  constructor(props) {
    super(props);
    this.state = {
      navigation: ListActionsNavigationList,
    }
  }

  changeView = (e, item) => {
    e.preventDefault();
    this.props.changeView(item.component, item.title)
  };

  render() {
    const { navigation } = this.state;
    return (
      <List>
        {navigation.map((item, index) => (
          <Item key={index} >
            <Link onClick={(e) => this.changeView(e, item)}>
              {item.title}
            </Link>
          </Item>
        ))}
      </List>
    );
  }
}

export default ListActionsNavigation;
