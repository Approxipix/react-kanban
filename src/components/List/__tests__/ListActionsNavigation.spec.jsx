import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import ListActionsNavigation from '../ListActions/ListActionsNavigation';

describe('<ListActionsNavigation>', () => {
  let component;
  let mockChangeView = jest.fn();

  beforeEach(() => {
    const state = {
      navigation: [
        { title: 'Move List', component: 'ListMove', },
        { title: 'Move All Cards', component: 'ListMoveAllCards', },
        { title: 'Delete List', component: 'ListDelete', },
      ]
    };

    component = shallow(<ListActionsNavigation changeView={mockChangeView} />);
    component.setState({ ...state });
  });

  it('should renders without crashing', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a nav link for each item from state', () => {
    expect(component.find('Link').length).toEqual(3)
  });

  it("should call changeView after click on nav item", () => {
    component.find('Link').first().simulate('click', {
      preventDefault: () => {}
    });

    expect(mockChangeView).toHaveBeenCalledTimes(1);
    expect(mockChangeView).toHaveBeenCalledWith('ListMove', 'Move List');
  });
});
