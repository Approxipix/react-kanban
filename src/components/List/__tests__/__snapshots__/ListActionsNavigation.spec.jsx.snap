// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListActionsNavigation> should renders without crashing 1`] = `
<styled.ul>
  <styled.li
    key="0"
  >
    <Link
      onClick={[Function]}
    >
      Move List
    </Link>
  </styled.li>
  <styled.li
    key="1"
  >
    <Link
      onClick={[Function]}
    >
      Move All Cards
    </Link>
  </styled.li>
  <styled.li
    key="2"
  >
    <Link
      onClick={[Function]}
    >
      Delete List
    </Link>
  </styled.li>
</styled.ul>
`;
